### PR TITLE
Fix general excepts in utilities

### PR DESCRIPTION
**Pull request by lfdversluis**

### CODE REVIEW

**Comment by lfdversluis** [comment]
```
Regression

Tribler.Test.test_remote_search.TestRemoteTorrentSearch.test_remote_torrent_search (from nosetests)
Failing for the past 1 build (Since Failed#2138 )
Took 40 sec.
Error Message

Test raised 1 unhandled Twisted exceptions, last one was: Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/RemoteTorrentHandler.py", line 478, in _success_callback
    self._remote_torrent_handler.save_torrent(tdef)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/util.py", line 37, in helper
    return func(*args, **kargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/RemoteTorrentHandler.py", line 168, in save_torrent
    self.torrent_db.addExternalTorrent(tdef, extra_info={u"is_collected": 1, u"status": u"good"})
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 305, in addExternalTorrent
    self._addTorrentToDB(torrentdef, extra_info)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 407, in _addTorrentToDB
    database_dict = self._get_database_dict(torrentdef, extra_info)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 388, in _get_database_dict
    "category": self.category.calculateCategory(torrentdef.metainfo, torrentdef.get_name_as_unicode()),
exceptions.AttributeError: 'NoneType' object has no attribute 'calculateCategory'
```

**Comment by lfdversluis** [comment]
retest this please

**Comment by devos50** [comment]
@lfdversluis fyi, that error is probably caused by the `@call_on_reactor_thread` callback above `_success_callback` and `save_torrent` in `RemoteTorrentHandler` (the call is scheduled and executed after the shutdown procedure is done). Making this a blocking call on the reactor thread solves the problem but I'm not sure about the consequences of this. Do you think this is a good idea and what possible consequences of this are?

Anyway, after the threaded reactor is gone, this error should not occur anymore I think?

**Comment by lfdversluis** [comment]
```
======================================================================
ERROR: test_channel_search (Tribler.Test.test_remote_search.TestRemoteChannelSearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Test/test_as_server.py", line 642, in tearDown
    AbstractServer.tearDown(self, annotate=False)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 157, in g
    d = func(*args, **kargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Test/test_as_server.py", line 167, in tearDown
    process_unhandled_twisted_exceptions()
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Test/util.py", line 130, in check_exceptions
    % (num_twisted_exceptions, self._twisted_exceptions[-1]['log_text']))
Exception: Test raised 2 unhandled Twisted exceptions, last one was: Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 825, in runUntilCurrent
    call.func(*call.args, **call.kw)
  File "<string>", line 2, in _do_request
    
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/RemoteTorrentHandler.py", line 32, in pass_when_stopped
    return f(self, *argv, **kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/RemoteTorrentHandler.py", line 455, in _do_request
    trackers = self._torrent_db_handler.getTrackerListByInfohash(infohash)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 696, in getTrackerListByInfohash
    torrent_id = self.getTorrentID(infohash)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 242, in getTorrentID
    return self.getTorrentIDS([infohash, ]).get(infohash)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 262, in getTorrentIDS
    torrents = self._db.fetchall(sql_stmt, to_select)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/util.py", line 46, in helper
    return blockingCallFromThread(reactor, func, *args, **kargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/util.py", line 225, in blockingCallFromThread
    return f(*args, **kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 401, in fetchall
    res = self.execute_read(sql, args)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 310, in execute_read
    return self.execute(sql, args)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/util.py", line 46, in helper
    return blockingCallFromThread(reactor, func, *args, **kargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/util.py", line 225, in blockingCallFromThread
    return f(*args, **kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 263, in execute
    cur = self.get_cursor()
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/CacheDB/sqlitecachedb.py", line 196, in get_cursor
    self._cursor_table[thread_name] = self._connection.cursor()
exceptions.AttributeError: 'NoneType' object has no attribute 'cursor'
```

**Comment by devos50** [comment]
@lfdversluis also, I will take a look at that error and try to fix it after lunch since that one's getting quite annoying...

**Comment by lfdversluis** [comment]
@devos50 Isn't the remotetorrenthandler called on the reactor thread already? if it is, making it a blocking call wouldn't solve much then because it's just executed then anyway without any blocking behaviour.

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
```
Regression

Tribler.Test.Core.test_session.TestSessionAsServer.test_add_torrent_def_to_channel (from nosetests)
Failing for the past 1 build (Since Failed#2141 )
Took 2.5 sec.
Error Message

Test raised 2 unhandled exceptions, last one was: PyDeadObjectError('The C++ part of the ActivityPanel object has been deleted, attribute access no longer allowed.',)
```

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
@devos50 Can you review? I don't see how I can cover the one line not covered in a test. If you have an idea let me know.

File: Tribler/Core/Utilities/utilities.py [hunk 1]
```diff
@@ -170,8 +170,8 @@ def isValidTorrentFile(metainfo):
     try:
         validTorrentFile(metainfo)
         return True
-    except:
-        print_exc()
+    except ValueError:
+        logger.exception("Could not check torrentfile: a ValueError was thrown")
```
Comment (devos50) [inline Tribler/Core/Utilities/utilities.py]: Please add a space between torrent and file

File: Tribler/Core/Utilities/tracker_utils.py [hunk 2]
```diff
@@ -102,10 +102,7 @@ def parse_tracker_url(tracker_url):
     # get port number if exists, otherwise, use HTTP default 80
     if hostname_part.find(u':') != -1:
         hostname, port = hostname_part.split(u':', 1)
-        try:
-            port = int(port)
-        except:
```
Comment (devos50) [inline Tribler/Core/Utilities/tracker_utils.py]: Why did you remove this exception?  Can it happen that the `port` is not valid?

**Comment by devos50** [comment]
@lfdversluis I gave my comments. That line is indeed very hard to cover without altering the method too much. I would leave it as it is.

File: Tribler/Core/Utilities/tracker_utils.py (hunk 2 above)
Comment (lfdversluis) [inline Tribler/Core/Utilities/tracker_utils.py]: I discussed this with @whirm. It first raised an Value error, which we would reraise. Might aswell let it throw then.

**Comment by lfdversluis** [comment]
@whirm ready for your review.

**Comment by lfdversluis** [comment]
```
Regression

Tribler.Test.test_libtorrent_download.TestLibtorrentDownload.test_downloadfromfile (from nosetests)
Failing for the past 1 build (Since Failed#2216 )
Took 27 sec.
Error Message

Test raised 1 unhandled exceptions, last one was: TypeError('__init__() takes exactly 2 arguments (1 given)',)
```

**Comment by lfdversluis** [comment]
retest this splease

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
```
Regression

Tribler.Test.Core.Libtorrent.test_libtorrent_download_impl.TestLibtorrentDownloadImpl.test_restart (from nosetests)
Failing for the past 1 build (Since Failed#2217 )
Took 0.13 sec.
Error Message

Couldn't listen on any:38972: [Errno 98] Address already in use.
```

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
8x
```

Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/endpoint.py", line 250, in dispersythread_data_came_in
    u"standalone_ep")
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/dispersy.py", line 1464, in on_incoming_packets
    (str(packet[0]), repr(packet[1])) for packet in packets]
exceptions.AssertionError: [('<Candidate 97.87.83.206:44990>', "'A\\x00\\xad\\x9d\\x1f\\xcb\\xf1!\\x00\\x00\\x00\\x00\\x00\\x10\\x00\\x00z9\\x00\\x00'")]
```

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
PyDeadObjectError again

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/RemoteTorrentHandler.py", line 478, in _success_callback
    self._remote_torrent_handler.save_torrent(tdef)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/dispersy/util.py", line 37, in helper
    return func(*args, **kargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/RemoteTorrentHandler.py", line 168, in save_torrent
    self.torrent_db.addExternalTorrent(tdef, extra_info={u"is_collected": 1, u"status": u"good"})
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 305, in addExternalTorrent
    self._addTorrentToDB(torrentdef, extra_info)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 407, in _addTorrentToDB
    database_dict = self._get_database_dict(torrentdef, extra_info)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@2/tribler/Tribler/Core/CacheDB/SqliteCacheDBHandler.py", line 388, in _get_database_dict
    "category": self.category.calculateCategory(torrentdef.metainfo, torrentdef.get_name_as_unicode()),
exceptions.AttributeError: 'NoneType' object has no attribute 'calculateCategory'
```

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
PyDeadObjectError again

**Comment by lfdversluis** [comment]
retest this please

**Comment by lfdversluis** [comment]
@whirm ready for your review.
